8.0 'mrp' (dev)

legalsylvain opened the pull request:
Port 'mrp' modification realized by OneStein BV. (branch onesteinbv:8.0-os)

legalsylvain committed:
[ADD] [8.0] 'mrp'. add pre-migration script; improve post-migration script;

hbrunn commented:
This migration should do the rename from https://github.com/OpenUpgrade/OpenUpgrade/pull/121#discussion_r20920489

hbrunn committed:
[ADD] rename xmlids from procurement module

hbrunn commented on addons/mrp/migrations/8.0.1.1/post-migrate.py:
```diff
@@ -1,23 +1,199 @@
+        # attribute_value_ids - new in 8.0
+        # TODO: property_ids
+    # Remove unneeded items
+    cr.execute("DELETE FROM mrp_bom WHERE id in (%s)" % ','.join(ids))
```
this breaks if ids is empty

hbrunn committed:
[FIX] don't break on empty list

hbrunn committed:
[FIX] we don't need to rename property access, they already exist in mrp
7.0 and will be updated to the new group during migration

hbrunn committed:
[FIX] cope with empty origin

vaab commented:
The code here seems far from finished. Some commented code remains everywhere. Can we have an update on what is the status of this branch ? It's tagged "Needs review" but seems more "Work in Progress".

Another very specific remark if this helps:

```
def update_stock_moves(cr, pool, uid):
    stock_move_obj = pool['stock.move']
    sm_ids = stock_move_obj.search(cr, uid, [])
    mrp_production_obj = pool['mrp.production']
    location_obj = pool['stock.location']
    location_id = location_obj.search(cr, uid, [('name', '=', 'Production')])
    for sm in stock_move_obj.browse(cr, uid, sm_ids):
        if 'MO' in sm.name and sm.location_dest_id.id == location_id[0]:
            prod_id = mrp_production_obj.search(cr, uid, [('name', '=', sm.name)])
            sql = """UPDATE stock_move SET raw_material_production_id = %s WHERE id = % s""" % (prod_id[0], sm.id)
            cr.execute(sql)
            cr.commit()
#             sm.write(cr, uid, {'raw_material_production_id': prod_id[0]})
```

This function:
- is far from optimal: the search domain of `stock.move` could replace the python "if".
- contains commented code (which can be removed)
- has a bug if no "mrp.production" object matches the search (and thus `prod_id` is empty). And that bug is happening on a customer database I'm trying to migrate.

Can you be more specific (by commenting the code for instance) onto what's going on ? I'm not very well into the business logic of these modules... so perhaps all this is obvious.

Thanks anyway for the huge work being done.

hbrunn commented:
all the *-onesteinbv branches are part of one big code dump that sylvain took the effort to break into several PRs. Given that the original authors seem to have abandoned this, we'll have to fix whatever is wrong. So please go ahead and make PRs.

Commit:
[IMP] Trigger creation of standard mrp procurement rules
[IMP] Migrate track_production on products

Commit:
[UPD] Copyright notices

Commit:
[RFR] Code cleanup

StefanRijnhart commented:
@vaab Definitely work in progress. I will be looking further into this one in the coming days.

Commit:
[ADD] Update the procurements' move lines

Commit:
[RFR] Cleanup migration of properties on bom lines

Commit:
[RFR] Take advantage of global uid variable
[ADD] Get message_last_post for the appropriate models

Commit:
[RFR] Query to set product template on BOM

hbrunn committed:
[FIX] pass cursor to logged_query

hbrunn committed:
[FIX] update correct model

ericnjogu commented:
Greetings, I applied this pull request and got this error - http://pastebin.com/nEL5L4RS (ProgrammingError: syntax error at or near "FROM")

legalsylvain commented:
@kunadawa I'll take a look in a while.
@StefanRijnhart, @hbrunn, except that bug, do you still consider this module as WIP ?

hbrunn committed:
[FIX] drop view only if it exists

thomaspaulb commented:
Like kunadawa, I got the same error:

```
post-migrate.py: syntax error at or near "FROM"
LINE 5:             FROM mrp_bom_mrp_property_rel bom_rel
```

To solve it, I needed to make the following change to post_migration.py:

```
INSERT INTO mrp_bom_line_mrp_property_rel
(mrp_bom_line_id, mrp_property_id)
VALUES(%s, bom_rel.mrp_property_id)
FROM mrp_bom_mrp_property_rel bom_rel
WHERE bom_rel.mrp_bom_id = %s
```

Into:

```
INSERT INTO mrp_bom_line_mrp_property_rel
(mrp_bom_line_id, mrp_property_id)
SELECT %s, bom_rel.mrp_property_id
FROM mrp_bom_mrp_property_rel bom_rel
WHERE bom_rel.mrp_bom_id = %s
```

hbrunn commented:
please provide a PR against this branch

Commit:
Resolves the following critical error in post-migrate.py,

post-migrate.py: syntax error at or near "FROM"
LINE 5:             FROM mrp_bom_mrp_property_rel bom_rel

thomaspaulb commented:
https://github.com/OpenUpgrade-dev/OpenUpgrade/pull/3

pedrobaeza committed:
Merge pull request #3 from antiflu/8.0-mrp-antiflu

Resolves the following critical error in post-migrate.py,

gunaidin commented:
Hello,
I have copied pre-migration.py and post-migrate.py into the main branch mrp module, and I get the following error (list index out of range):
Traceback (most recent call last):
........
  File "/OpenUpgrade-8.0/addons/mrp/migrations/8.0.1.1/post-migrate.py", line 105, in update_stock_moves
    sql = """UPDATE stock_move SET raw_material_production_id = %s WHERE id = % s""" % (prod_id[0], sm.id)
IndexError: list index out of range

pedrobaeza commented:
@gunaidin, this is being treated here: https://github.com/OpenUpgrade-dev/OpenUpgrade/pull/5.

legalsylvain commented:
TODO : 
https://github.com/OpenUpgrade/OpenUpgrade/pull/297/files#diff-eddf81f93c3a7e4031b064b5b071df47R12
(#297)

hbrunn committed:
[FIX] get production location from property

support multicompany

hbrunn commented:
@all please review https://github.com/OpenUpgrade-dev/OpenUpgrade/pull/5 and https://github.com/OpenUpgrade-dev/OpenUpgrade/pull/6 so that we don't need merges on the merge to have a working migration

pedrobaeza committed:
Merge pull request #6 from hbrunn/8.0-update_stock_moves

[FIX] get production location from property

pedrobaeza commented on addons/mrp/migrations/8.0.1.1/pre-migration.py:
```diff
@@ -0,0 +1,75 @@
+
+        prod_ids = tuple([x for x, in tuple(cr.fetchall())])
+
+        sql = """UPDATE mrp_production SET state = 'draft' WHERE id in {}""".format(prod_ids)
```
This fails if prod_ids has only one element, because it prints `(id,)`. You should:

```
sql = """UPDATE mrp_production SET state = 'draft' WHERE id in %s"""
cr.execute(sql, (prod_ids,))
```

pedrobaeza replied:
Better if we do a logged_query

pedrobaeza commented on addons/mrp/migrations/8.0.1.1/pre-migration.py:
```diff
@@ -0,0 +1,75 @@
+
+        sql = """UPDATE mrp_production SET state = 'draft' WHERE id in {}""".format(prod_ids)
+        cr.execute(sql)
+        cr.commit()
```
Don't make manual commits

pedrobaeza commented on addons/mrp/migrations/8.0.1.1/post-migrate.py:
```diff
@@ -1,23 +1,234 @@
+    sql = """UPDATE ir_act_window SET domain = NULL WHERE domain =
+'[(''bom_id'',''='',False)]' AND res_model = 'mrp.bom'"""
+    cr.execute(sql)
+    cr.commit()
```
Don't make manual commits

pedrobaeza commented on addons/mrp/migrations/8.0.1.1/post-migrate.py:
```diff
@@ -1,23 +1,234 @@
+def fix_domains(cr, pool):
+    sql = """UPDATE ir_act_window SET domain = NULL WHERE domain =
+'[(''bom_id'',''='',False)]' AND res_model = 'mrp.bom'"""
+    cr.execute(sql)
```
Better to do a logged_query

pedrobaeza committed:
[IMP] mrp: Some adjustments in scripts

sebalix commented:
I tested this PR, we had just some BoM to migrate regarding the `mrp` part, but all goes well.

pedrobaeza commented:
I think this has been enough in testing, so I merge and possible issues can be solved via new PRs.
